idseq_dag/steps/run_srst2.py

```diff
@@ -0,0 +1,126 @@
+import os
+import pandas as pd
+import shutil
+
+from idseq_dag.engine.pipeline_step import PipelineStep
+from idseq_dag.util.s3 import fetch_from_s3
+import idseq_dag.util.command as command
+
+class PipelineStepRunSRST2(PipelineStep):
+    '''
+    srst2 is used to detect resistance genes.
+    See: https://github.com/katholt/srst2
+    '''
+
+    def run(self):
+        ''' Invoking srst2 '''
+        OUTPUT_LOG = 'output.log'
+        OUTPUT_GENES = 'output__genes__ARGannot_r2__results.txt'
+        OUTPUT_FULL_GENES = 'output__fullgenes__ARGannot_r2__results.txt'
+        isPaired = (len(self.input_files_local[0]) == 2)
+        isFasta = (self.additional_attributes['file_type'] == 'fasta')
+        isZipped = (self.input_files_local[0][0][-3:] == '.gz')
+        self.execute_srst2(isPaired, isFasta, isZipped)
+        log = os.path.join(self.output_dir_local, OUTPUT_LOG)
+        log_dest = self.output_files_local()[0]
+        results = os.path.join(self.output_dir_local, OUTPUT_GENES)
+        results_dest = self.output_files_local()[1]
+        shutil.move(log, log_dest)
+        shutil.move(results, results_dest) 
+        if not os.path.exists(os.path.join(self.output_dir_local, OUTPUT_FULL_GENES)): 
+            for f in self.output_files_local()[2:5]:
+                PipelineStepRunSRST2.fill_file_path(f)
+        else:
+            # Post processing of amr data
+            results_full = os.path.join(self.output_dir_local, OUTPUT_FULL_GENES)
+            results_full_dest = self.output_files_local()[2]
+            shutil.move(results_full, results_full_dest)
+            self.process_amr_results(results_full_dest)
+
+    # Inherited method
+    def count_reads(self):
+        pass
+
+    def execute_srst2(self, isPaired, isFasta, isZipped):
+        """Executes srst2 with appropriate parameters based on whether input files are zipped,
+           paired reads and on file type."""        
+        srst2_params = ['srst2']
+        srst2_params.extend(self.get_common_params())
+        if isFasta: 
+            file_ext = '.fasta.gz' if isZipped else '.fasta'
+            srst2_params.extend(['--read_type', 'f'])
+        else: 
+            file_ext = '.fastq.gz' if isZipped else '.fastq'
```

Comment:
we also support fq

Reply:
yes, I'm renaming these files to the appropriate extensions in this dag step so it is accounted for